test/plugin_functional/test_suites/core_plugins/status.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import expect from '@kbn/expect';
+import { ServiceStatusLevels } from '../../../../src/core/server';
+import { PluginFunctionalProviderContext } from '../../services';
+
+export default function ({ getService, getPageObjects }: PluginFunctionalProviderContext) {
+  const supertest = getService('supertest');
+  const log = getService('log');
+
+  const delay = (ms: number) => new Promise((r) => setTimeout(r, ms));
+  const getStatus = async (pluginName?: string) => {
+    const resp = await supertest.get('/api/status?v8format=true');
+
+    if (pluginName) {
+      return resp.body.status.plugins[pluginName];
+    } else {
+      return resp.body.status.overall;
+    }
+  };
+
+  const setStatus = async <T extends keyof typeof ServiceStatusLevels>(level: T) =>
+    supertest
+      .post(`/internal/core_plugin_a/status/set?level=${level}`)
+      .set('kbn-xsrf', 'xxx')
+      .expect(200);
+
+  describe('status service', () => {
+    // This test must comes first because the timeout only applies to the initial emission
+    it("returns a timeout for status check that doesn't emit after 30s", async () => {
+      let aStatus = await getStatus('corePluginA');
+      expect(aStatus.level).to.eql('unavailable');
+
+      // Status will remain in unavailable due to core services until custom status timesout
+      // Keep polling until that condition ends, up to a timeout
+      const start = Date.now();
+      while ('elasticsearch' in (aStatus.meta?.affectedServices ?? {})) {
```

Comment:
This `while` condition is a bit obscure to me, even with the comment. When the timeout status kicks in, we no longer have `aStatus.meta?.affectedServices.elasticsearch` ?

Reply:
I actually probably need to re-think how to implement this test, because I don't think this will be compatible with the changes in #102108 since Kibana will never become available. I think I'll:
- Change the `wait` message on this FTR suite to the `http is running` one. I may end up moving this to a separate suite / config to be sure it doesn't break other tests.
- Just poll until the status for `corePluginA` has the timeout message.